app/components/post_body/index.js

```diff
@@ -26,6 +41,25 @@ function mapStateToProps(state, ownProps) {
         isPending = false;
     }
 
+    const channel = getCurrentChannel(state);
+    const user = getCurrentUser(state);
+    const teamMember = getCurrentTeamMembership(state);
+    const channelMember = getMyCurrentChannelMembership(state);
+    const {config, license} = state.entities.general;
+    const isUserCanManageMembers = canManageMembers(channel, user, teamMember, channelMember, config, license);
+    const isEphemeralPost = isPostEphemeral(post);
+
+    let isPostAddChannelMember = false;
+    if (
+        (channel.type === General.PRIVATE_CHANNEL || channel.type === General.OPEN_CHANNEL) &&
```

Comment:
I don't understand all this logic. Why can't we just check the post type?

Reply:
The post type here is `system_ephemeral` which is commonly used.
I'll try to rearrange the logic which will only do things if necessary. Also, I realized that some checks were already covered by `canManageMembers`.

Reply:
Oh, I see. We should really separate the notion of ephemeral posts from post type at some point so that we can give a type to ephemeral posts. Could we just look at the post props then and skip the other checks? This message shouldn't be created for non-private/public channels anyway